src/run.js

```diff
@@ -405,15 +405,18 @@ const processPage = ({
 
 /**
  *
- * @param {{ urls: Array<string>, debug: boolean, loadimages: boolean, skippable: function, browser: any, userAgent: string, withoutjavascript: boolean, viewport: any, puppeteerArgs: Array<string>, cssoOptions: Object, ignoreCSSErrors?: boolean, ignoreJSErrors?: boolean, styletags?: boolean, enableServiceWorkers?: boolean, disableJavaScript?: boolean }} options
+ * @param {{ urls: Array<string>, debug: boolean, loadimages: boolean, skippable: function, browser: any, userAgent: string, withoutjavascript: boolean, viewport: any, puppeteerArgs: Array<string>, cssoOptions: Object, ignoreCSSErrors?: boolean, ignoreJSErrors?: boolean, styletags?: boolean, enableServiceWorkers?: boolean, disableJavaScript?: boolean, whitelist?: Array<string> }} options
  * @return Promise<{ finalCss: string, stylesheetContents: { [key: string]: string } }>
  */
 const minimalcss = async options => {
-  const { urls } = options;
+  const { urls, whitelist = [] } = options;
   const debug = options.debug || false;
   const cssoOptions = options.cssoOptions || {};
   const enableServiceWorkers = options.enableServiceWorkers || false;
   const puppeteerArgs = options.puppeteerArgs || [];
+  const isInWhiteList = selector =>
+    whitelist.some(rule => new RegExp(rule).test(selector));
+
```

Comment:
I don't see the point of this being a standalone function. It's not curried either. And maybe that's the key here. 

If you have more than 0 whitelist selectors, and you're processing 1,000 selectors you have to execute `new RegExp(whitelist[i])` 1,000 times. And if you have 3 whitelist selectors you might be executing `new RegExp(whitelist[i])` 3,000 times. Or am I missing something?

A simple curry could be to make an array of regexes instead. Perhaps:
```javascript
const whitelistRules = whitelist.map(rule => new RegExp(rule));

... // somewhere around line 504

if (whitelistRules.some(regex => regex.test(selectorString))) {
  return true;
```

Reply:
Good point! Thank you!